File: movieApp/theatre/src/main/java/com/example/theatre/service/MovieService.java
package com.example.theatre.service;

import com.example.theatre.domain.*;
import com.example.theatre.exception.MovieAlreadyExistsException;
import com.example.theatre.exception.TheatreAlreadyExistsException;
import com.example.theatre.exception.TheatreNotFoundException;

import java.util.List;

public interface MovieService {
    Movie addMovie(Movie movie) throws MovieAlreadyExistsException;
    Movie addTheatre(MovieTheatre movieTheatre,String movieID);
    Movie getMovieByID(String movieID);
    List<Movie> getMoviesByCity(String city);
    List<Movie> getMoviesByCityAndLanguage(String city,String language);
    List<Movie> getMoviesByCityAndGenre(String city,String genre);
    List<Movie> getMoviesByCityAndFormat(String city,String format);
    Movie updateMovieStatus(String movieID,String status);
    List<Movie> getMoviesByCityAndUpcoming(String city);
    List<Theatre> getTheatresByMovieID(String movieID,String city);
    List<Theatre> getTheatresByMovieIDAndFormat(String movieID,String format,String city) throws TheatreNotFoundException;
    List<Theatre> getTheatresByMovieIDAndLanguage(String movieID,String language,String city) throws TheatreNotFoundException;
    Movie addShowtime(ShowScreen showScreen, String movieID, String theatreID);
    Movie deleteShowtime(ShowScreen showScreen,String movieID,String theatreID);
    int getTicketPrice(ScreenDateTime screenDateTime,String theatreID,String movieID,String format);
}

File: src/app/services/movie.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Movie } from '../model/movie';
import { Theatre } from '../model/theatre';

@Injectable({
  providedIn: 'root'
})
export class MovieService {
  moviesURL:string="http://localhost:3333/movieControl";
  constructor(private httpClient:HttpClient) { }
  getMoviesByCity(city:string){
      return this.httpClient.get<Movie[]>(this.moviesURL+"/getMoviesByCity/"+city);
  }
  getMoviesByCityAndLanguage(city:string,language:string){
    return this.httpClient.get<Movie[]>(this.moviesURL+"/getMoviesByCityAndLanguages/"+city+"/"+language);
  }
  getMoviesByCityAndGenre(city:string,genre:string){
    return this.httpClient.get<Movie[]>(this.moviesURL+"/getMoviesByCityAndGenre/"+city+"/"+genre);
  }
  getMoviesByCityAndFormat(city:string,format:string){
    return this.httpClient.get<Movie[]>(this.moviesURL+"/getMoviesByCityAndFormat/"+city+"/"+format);
  }
  getMoviesByCityAndUpcoming(city:string){
    return this.httpClient.get<Movie[]>(this.moviesURL+"/getMoviesByCityAndUpcoming/"+city);
  }
  getMovieByID(movieID:string){
    return this.httpClient.get<Movie>(this.moviesURL+"/getMovieByID/"+movieID);
  }
  getTheatresByMovieID(movieID:string,city:string){
    return this.httpClient.get<Theatre[]>(this.moviesURL+"/getTheatresByMovieID/"+movieID+"/"+city);
  }
  getAmount(obj:any,theatreID:string,movieID:string,format:string){
    return this.httpClient.post<number>(this.moviesURL+"/getAmount/"+theatreID+"/"+movieID+"/"+format,obj);
  }
}
